admin_panel/exam_batch/model.py

```diff
@@ -21,20 +57,290 @@ class ExamBatch(models.Model):
         on_delete=models.CASCADE,
         related_name='exam_batches'
     )
-    start_date = models.DateField(null=True, blank=True)
-    end_date = models.DateField(null=True, blank=True)
-
     organization = models.ForeignKey(
         'Organization',
         on_delete=models.CASCADE,
         related_name='exam_batches'
     )
-    created_at = models.DateTimeField(auto_now_add=True)
-    updated_at = models.DateTimeField(auto_now=True)
+    mode = models.CharField(
+        max_length=50,
+        choices=ExamMode.choices,
+        help_text="Type of examination mode"
+    )
+    # Basic schedule
+    start_date = models.DateField()
+    end_date = models.DateField()
+    start_time = models.TimeField()
+    end_time = models.TimeField()
+    # Evaluation schedule
+    evaluation_start_date = models.DateField(null=True, blank=True)
+    evaluation_end_date = models.DateField(null=True, blank=True)
+    evaluation_start_time = models.TimeField(null=True, blank=True)
+    evaluation_end_time = models.TimeField(null=True, blank=True)
```

Comment:
<div>


<div id="suggestion">
<div id="issue"><b>Consider adding date-time field validation</b></div>
<div id="fix">

Consider adding validation for date and time fields to ensure `start_date` is before `end_date`, and `start_time` is before `end_time`. This could help prevent potential logical errors in exam scheduling.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -79,3 +79,12 @@ class ExamBatch(models.Model):
      evaluation_start_time = models.TimeField(null=True, blank=True)
      evaluation_end_time = models.TimeField(null=True, blank=True)
 +
 +    def clean(self):
 +        if self.start_date and self.end_date and self.start_date > self.end_date:
 +            raise ValidationError('Start date must be before end date')
 +        if self.start_time and self.end_time and self.start_time > self.end_time:
 +            raise ValidationError('Start time must be before end time')
 +        if self.evaluation_start_date and self.evaluation_end_date and self.evaluation_start_date > self.evaluation_end_date:
 +            raise ValidationError('Evaluation start date must be before evaluation end date')
 +        if self.evaluation_start_time and self.evaluation_end_time and self.evaluation_start_time > self.evaluation_end_time:
 +            raise ValidationError('Evaluation start time must be before evaluation end time')
```
</div>
</details>
</div>


<small><i>Code Review Run <a href=https://github.com/sleep-404/ExamEvoke/pull/10#issuecomment-2499983195>#fd6475</a></i></small>
</div>

---
Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged